pkg/plugins/generic/generic_plugin.go

```diff
@@ -197,3 +203,24 @@ func isCommandNotFound(err error) bool {
 	}
 	return false
 }
+
+func handleSriovModeStateChange(iface *sriovnetworkv1.Interface, old *sriovnetworkv1.SriovNetworkNodeState) (needDrain bool, err error) {
+	err = nil
+	needDrain = false
+	if iface.SriovMode == "" {
+		if iface.SriovMode != "legacy" && iface.SriovMode != "switchdev" {
+			return false, fmt.Errorf("generic-plugin OnNodeStateChange(): not supported SR-IOV mode %s for "+
+				"device %s, supported modes [legacy, switchdev]", iface.PciAddress, iface.SriovMode)
+		}
+		if iface.NumVfs <= 0 {
+			return false, fmt.Errorf("generic-plugin OnNodeStateChange(): " +
+				"To change SR-IOV mode number of VFs should be > 0")
+		}
+		for _, ifaceOldStatus := range old.Status.Interfaces {
+			if ifaceOldStatus.PciAddress == iface.PciAddress && ifaceOldStatus.SriovMode != iface.SriovMode {
+				needDrain = true
+			}
+		}
+	}
+	return
```

Comment:
return needDrain, err ?

Reply:
return values already defined in the function header, it is something supported by go language

Reply:
i see, thx for clarifying